mlir/lib/Conversion/LLVMCommon/Pattern.cpp

```diff
@@ -106,8 +106,14 @@ bool ConvertToLLVMPattern::isConvertibleAndHasIdentityMaps(
 
 Type ConvertToLLVMPattern::getElementPtrType(MemRefType type) const {
   auto addressSpace = getTypeConverter()->getMemRefAddressSpace(type);
-  if (failed(addressSpace))
+  if (failed(addressSpace)) {
+    emitError(UnknownLoc::get(type.getContext()),
+              "conversion of memref memory space ")
+        << type.getMemorySpace()
+        << " to integer address space "
+           "failed. Consider adding memory space conversions.";
```

Comment:
This error should ideally be emitted on the operation that holds the type. With this, you can also improve your test to check that this was emitted on the correct line.

Reply:
Thanks, made edits to emit the error on the operation